tensorflow_estimator/python/estimator/exporter.py

```diff
@@ -293,15 +294,20 @@ def export(self, estimator, export_path, checkpoint_path, eval_result,
       self._best_eval_result = self._get_best_eval_result(
           full_event_file_pattern)
 
-    if self._best_eval_result is None or self._compare_fn(
-        best_eval_result=self._best_eval_result,
-        current_eval_result=eval_result):
+    if (self._best_eval_result is None or
+        # check if this is the first export. Note that
+        # evaluation occurs before exporting models and hence the loaded
+        # _best_eval_result can be the same even when no export had been done.
+        (not self._has_exported and self._best_eval_result == eval_result) or
```

Comment:
Thanks for the PR! Is there a reason why we need `self._best_eval_result == eval_result`? I would think that if it's not exported yet, i.e. `not self._has_exported`, is a sufficient condition.

Reply:
Consider a training job which has been stopped and it was resumed. In the first export, the value of `self._has_exported` is False and regardless of the first checkpoint is better or not, it will be exported which is not desirable.  
`self._best_eval_result == eval_result` avoids exporting in this scenario. This has been captured in `test_best_exporter_with_preemption` and this test will fail if `self._best_eval_result == eval_result` is removed.

Reply:
Thanks, in that case should we make the condition "not self._has_exported and eval_result better than or equal to self._best_eval_result" as opposed to ==?

Reply:
Sure, this is correct. 

Current implementation reflects the same logic:
`(not self._has_exported and self._best_eval_result == eval_result) or self._compare_fn(best_eval_result=self._best_eval_result, current_eval_result=eval_result))`.

Note that `self._best_eval_result` and `eval_result` are a `dict` and should be compared by `self._compare_fn`. Also if the `eval_result` is better than `self._best_eval_result`, regardless of it was imported or not before, the model should be exported.